R/config.R

```diff
@@ -241,25 +241,30 @@ read_json_into_config <- function(config_path, optional_fields) {
   }
 
   inner <- function(raw_data, class_to_fill) {
-    # For each property, check if it is a regular value, or an S7 object.
-    # If it is an S7 object, we need to create an instance of that class, and do
-    # all the same checks for properties that we did above. If not, just add it
-    # to the config object.
+    # Helper to convert list(NULL) or list() to NULL recursively
+    nullify_list_null <- function(x) {
```

Comment:
Maybe we could add a comment in the code about what sort of data cases require this function. It's just not quite clear to me what case this covers, and why it is needed if we're just adding test coverage. Makes me slightly anxious about if it will break anything in production.

Also, can we add back in the comment that got deleted? That was helpful in understanding what this recursive function does.